checks/raw/pinned_dependencies.go

```diff
@@ -67,14 +73,87 @@ func PinningDependencies(c *checker.CheckRequest) (checker.PinningDependenciesDa
 		return checker.PinningDependenciesData{}, err
 	}
 
-	if unpinnedNugetDependencies := getUnpinnedNugetDependencies(&results); len(unpinnedNugetDependencies) > 0 {
-		if err := processCsprojLockedMode(c, unpinnedNugetDependencies); err != nil {
-			return checker.PinningDependenciesData{}, err
-		}
+	// Nuget Post Processing
+	if err := postProcessNugetDependencies(c, &results); err != nil {
+		return checker.PinningDependenciesData{}, err
 	}
+
 	return results, nil
 }
 
+func postProcessNugetDependencies(c *checker.CheckRequest,
+	pinningDependenciesData *checker.PinningDependenciesData,
+) error {
+	if unpinnedDependencies := getUnpinnedNugetDependencies(pinningDependenciesData); len(unpinnedDependencies) > 0 {
+		var nugetPostProcessData NugetPostProcessData
+		if err := retrieveNugetCentralPackageManagement(c, &nugetPostProcessData); err != nil {
+			return err
+		}
+		if err := retrieveCsprojConfig(c, &nugetPostProcessData); err != nil {
+			return err
+		}
+		if nugetPostProcessData.CpmConfig.IsCPMEnabled {
+			collectPostProcessNugetCPMDependencies(unpinnedDependencies, &nugetPostProcessData)
+		} else {
+			collectPostProcessNugetCsprojDependencies(unpinnedDependencies, &nugetPostProcessData)
+		}
+	}
+	return nil
+}
+
+func collectPostProcessNugetCPMDependencies(unpinnedNugetDependencies []*checker.Dependency,
+	postProcessingData *NugetPostProcessData,
+) {
+	packageVersions := postProcessingData.CpmConfig.PackageVersions
+
+	numFixedVersions, unfixedVersions := countFixedVersions(packageVersions)
+	// if all dependencies are fixed to specific versions, pin all dependencies
+	if numFixedVersions == len(packageVersions) {
+		pinAllNugetDependencies(unpinnedNugetDependencies)
+	} else {
+		// if some or all dependencies are not fixed to specific versions, update the remediation
+		for i := range unpinnedNugetDependencies {
+			(unpinnedNugetDependencies)[i].Remediation.Text = (unpinnedNugetDependencies)[i].Remediation.Text +
+				": some of dependency versions are not fixes to specific versions: " + unfixedVersions
+		}
+	}
+}
+
+func retrieveNugetCentralPackageManagement(c *checker.CheckRequest, nugetPostProcessData *NugetPostProcessData) error {
+	if err := fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
+		Pattern:       "Directory.*.props",
```

Comment:
it does, but the product group decided to allow this use other types of props file, since folks (and older versions of nuget) use/allow usage of other props file to declare dependencies such as Directory.Build.pros and others.